sdk/src/wallet/wallet/operations/stronghold.rs

```diff
@@ -36,10 +39,23 @@ impl Wallet {
     }
 
     /// Stores a mnemonic into the Stronghold vault
-    pub async fn store_mnemonic(&self, mnemonic: impl Into<Mnemonic> + Send) -> crate::wallet::Result<()> {
+    pub async fn store_mnemonic<T>(&self, mnemonic: T) -> crate::wallet::Result<()>
```

Comment:
Kind of don't see the point of this. Should probably just force the caller to call `TryInto`.

Reply:
The point was to keep the API as convenient as it is currently, because that's what people may be used to. So my intention was to break the API only when I thought it was absolutely necessary to improve security for the user (like forcing ownership of the mnemonic string).

Reply:
This is a breaking change anyway though, as `impl Into<Mnemonic>` != `impl TryInto<Mnemonic>`.

Reply:
You could add `MnemonicLike`

Reply:
Ah, the change displayed above the comment is misleading. Still, I think it would be better to have a `MnemonicLike` that can apply to either a string or a mnemonic type.

Reply:
okay, I look into it! :+1: